### ISSUE TITLE
[AutoML-notebook] Why AUC is different (0.5) when trying manually compared to AutoML output's metric (0.63)

### ISSUE THREAD

**Issue by CESARDELATORRE** [Azure-Samples/AzureMLWorkshop#30]
Why AUC is different (0.5) when trying manually compared to AutoML output's metric for the best model (0.63)?, taking into account that the same validation dataset was provided to AutoMLConfig.

Need to follow up on this issue from a Data Science side:

AutoML output (METRIC is AUC):

```
ITERATION   PIPELINE                                       DURATION      METRIC      BEST
         0   MaxAbsScaler LightGBM                          0:02:33       0.6343    0.6343
         1   MinMaxScaler LightGBM                          0:02:18       0.6266    0.6343
```

Your own metric (AUC) calculation using the same validation dataset:

AUC: 0.5

Need to review the notebook code and fix it if the calculation is performed wrong.

**Comment by CESARDELATORRE** [comment]
Related/same issue than this: https://github.com/Azure-Samples/AzureMLWorkshop/issues/25

**Comment by CESARDELATORRE** [comment]
Researching this with Anup.
The optimization metric in AutoML is 'weighted' while in plain Scikit-Learn is not.
We might need to measure differently with our predictions at the end of the notebook..

**Comment by CESARDELATORRE** [comment]
Found the issue. The notebooks need to be updated, though. This is the issue's explanation:
 
AUC needs to be calculated with the probabilities, not the predictions.
Interestingly (Looks like a bug/issue in LightGBM, see: https://github.com/microsoft/LightGBM/issues/198#issuecomment-489419106), the original plain LightGBM was returning probabilities when using 'gbdt' when predicting instead of 0 and 1, even when the 'objective' is 'binary', not regression...:

![image](https://user-images.githubusercontent.com/1712635/76247832-6e196680-61fd-11ea-9683-fe41065fafb1.png)

![image](https://user-images.githubusercontent.com/1712635/76247914-8d17f880-61fd-11ea-9f98-dcc46546b54c.png)

Also, related to the issue/bug above from LightGBM, when using 'gbdt' you cannot call the method to get the predictions, as I get this error. That's a bug, since the objective is 'binary' classification:

![image](https://user-images.githubusercontent.com/1712635/76247962-a4ef7c80-61fd-11ea-9cc5-5abfcdd0df13.png)

That's why when calculating AUC it was "right", because it was obtaining probabilities, not predictions (0/1):

![image](https://user-images.githubusercontent.com/1712635/76247986-ad47b780-61fd-11ea-9fb4-39d5279464f4.png)

The original LightGBM notebook uses 'Booster' because it was the one who won 'Porto Seguro Kaggle competition':
https://github.com/microsoft/LightGBM/issues/198#issuecomment-489419782


But the predictions when using the AutoML model return 0 or 1, which is more logical, since this is a binary classification:

![image](https://user-images.githubusercontent.com/1712635/76248077-d9fbcf00-61fd-11ea-8b57-c061fac7490c.png)


Therefore we need to obtain the probabilities in an explicit way:

![image](https://user-images.githubusercontent.com/1712635/76248084-e2540a00-61fd-11ea-82bc-c317dbe29183.png)


Then AUC is calculated right when providing the probabilities column:

![image](https://user-images.githubusercontent.com/1712635/76248098-e849eb00-61fd-11ea-8b67-59bdbe9263fa.png)

It looks that it doesn't matter if it is weighted or not, in this case.
